sklearn/metrics/_ranking.py

```diff
@@ -856,7 +856,7 @@ def precision_recall_curve(y_true, probas_pred, *, pos_label=None, sample_weight
 
     precision = tps / (tps + fps)
     precision[np.isnan(precision)] = 0
-    recall = tps / tps[-1]
+    recall = np.ones(tps.size) if tps[-1] == 0 else tps / tps[-1]
```

Comment:
In [precision_recall_fscore_support](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_fscore_support.html?highlight=precision_recall_fscore_support#sklearn-metrics-precision-recall-fscore-support), there is a `zero_division` parameter that controls how we handle the edge case. This parameter is also used in `recall_score` and `precision_score`. If we want to be consistent with `precision_recall_fscore_support`, we would need to add a `zero_division` and use the same semantics.

What do you think?

Reply:
@thomasjpfan this issue has been sitting unfixed for years. I think this fix should be merged as-is before it bit rots, and open a separate issue for semantic consistency.

Reply:
I agree with @crypdick on this. It's been over 3 years for what is a very simple bugfix, and we can track semantic consistency via another issue. That issue can be tackled by someone else, potentially from the sklearn team, which would get the ball rolling sooner.

Reply:
Okay, let's keep the recall set to one. May we update `precision_recall_curve`'s docstring to reflect this behavior?